src/lib.rs

```diff
@@ -25,14 +25,24 @@ mod macros;
 pub mod array;
 pub mod number;
 pub mod object;
+
+#[cfg(not(feature = "thread_unsafe"))]
 pub mod string;
+#[cfg(not(feature = "thread_unsafe"))]
```

Comment:
Duplicating the entire `string` module seems like an unnecessary maintenance burden, when all we really need to do is change whether or not the various types are Send + Sync, plus change the container used to store the string cache - we could do that with fine-grained cfg statements.

Reply:
it felt I had to change too many places in the String code. 
Including:
```
    rc: AtomicUsize,
```

```
    static ref STRING_CACHE: DashSet<WeakIString> = DashSet::new();
```

Reply:
I don't think that's too many places.